pkg/tsdb/prometheus/querydata/response.go

```diff
@@ -123,7 +123,7 @@ func addMetadataToMultiFrame(q *models.Query, frame *data.Frame, enableDataplane
 		frame.Fields[1].Config = &data.FieldConfig{DisplayNameFromDS: customName}
 	}
 
-	if enableDataplane {
+	if enableDataplane && customName == "" {
```

Comment:
Is double equal intentional here?

Reply:
Yes, in the dataplane transformation support work, we decided that we would only override the name if legend format was set to auto. When the legend is set to auto, `getName` returns "". We want to check that customName is ""... but this could also work if we check that `q.LegendFormat == legendFormatAuto`

https://github.com/grafana/grafana/blob/bb5c052264f5ebdd8b649c8e879aeac64ca84ad9/pkg/tsdb/prometheus/querydata/response.go#L183-L191

Reply:
Damn! That's my bad. I thought it is a `.ts` file. I should not do PR reviews that late.